### PR TITLE
[WIP] issue #931: Rename core module from `evm` to `eth`

### PR DESCRIPTION
**Pull request by mneedham91**
### What was wrong?

"The core module used in py-evm is evm. This name is becoming less appropriate and if it is ever going to be renamed, now is the time." - @pipermerriam 

### How was it fixed?

So far:

- the evm package has been renamed to eth
- all references within eth (formerly evm) have been corrected
- all references within docs have been corrected

To do:

1. clean up (remove) pycharm files
2. inspect p2p, scripts/benchmark, stubs, tests, trinity for references to be corrected

### CODE REVIEW

File: eth/db/chain.py [hunk 1]
```diff
@@ -29,42 +29,42 @@
     to_tuple,
 )
 
-from eth_hash.auto import keccak
+from eth_hash.evm.auto import keccak
```
Comment (pipermerriam) [inline eth/db/chain.py]: I think this is an error.  Doesn't appear to be an `eth_hash.evm` module.

Comment (mneedham91) [reply eth/db/chain.py]: Yup, thanks!

**Comment by mneedham91** [comment]
Hi @pipermerriam have I botched something causing all these tests to fail? To be honest I'm not sure what those tests are doing.

**Comment by mneedham91** [comment]
Hi @pipermerriam did I botch something causes all these tests to fail? To be honest I'm not sure what the tests are doing.

**Comment by carver** [comment]
One of the toughest parts of this task is to keep rebasing against the latest master. The tests will fail if there is a merge conflict, IIRC.

**Comment by pipermerriam** [comment]
@mneedham91 I went ahead and did the rebase.  Yous should force pull from your master branch to get the updates.

**Comment by pipermerriam** [comment]
@mneedham91 thanks for doing the heavy lifting on this one.  I chose to go ahead and push it across the finish line to prevent the need for more rebasing as other things got merged.

**Pull request merged**